moon/cfc_77.15.580/server/net.moon

```diff
@@ -10,43 +10,71 @@ import flaggedMessages,
+    Section580.Alerter\alertDiscord steamId, nick, ip, Section580.netSpamThreshold, strName, spamCount
+
+-- Returns whether to ignore the message
 tallyUsage = ( message, ply, plySteamId, plyNick, plyIP ) ->
     return if Section580.safeNetMessages[message]
 
-    Section580.netSpam[plySteamId] or= {}
-    Section580.netSpam[plySteamId][message] or= 0
-    Section580.netSpam[plySteamId][message] += 1
+    Section580.netSpam[plySteamId] or= {
+        total: 0
+        messages: {}
+    }
 
-    spamCount = Section580.netSpam[plySteamId][message]
+    Section580.netSpam[plySteamId].messages[message] or= 0
+    Section580.netSpam[plySteamId].messages[message] += 1
+    Section580.netSpam[plySteamId].total += 1
 
+    totalCount = Section580.netSpam[plySteamId].total
+    spamCount = Section580.netSpam[plySteamId].messages[message]
+
+    -- Extreme spam for specific message
     if spamCount > Section580.netExtremeSpamThreshold
         alertMessage = "Player spamming many network messages! #{plyNick} (#{plySteamId}) is spamming: '#{strName}' (Count: #{spamCount} per #{Section580.netClearTime} seconds)"
         warnLog alertMessage
 
-        Section580.Alerter\alertStaff plySteamId, plyNick, strName, "extreme"
-        Section580.Alerter\alertDiscord plySteamId, plyNick, plyIP, Section580.netSpamThreshold, strName, spamCount
+        sendAlert plySteamId, plyNick, plyIP, Section580.netSpamThreshold, spamCount, "extreme"
+        bootPlayer ply
 
-        kickReason = "Suspected malicious action"
+        return true
 
-        if Section580.netShouldBan and plyIsValid
-            if ULib
-                ULib.ban ply, 1, kickReason
-            else
-                ply\Kick kickReason
+    -- Extreme spam for all messages
+    if totalCount > Section580.netTotalSpamThreshold
+        alertMessage = "Player spamming many network messages! #{plyNick} (#{plySteamId}) is spamming: #{totalCount} messages per #{Section580.netClearTime} seconds"
+        warnLog alertMessage
 
-        return
+        sendAlert plySteamId, plyNick, plyIP, Section580.netSpamThreshold, spamCount, "extreme"
+        bootPlayer ply
+
+        return true
 
+    -- Likely spam for specific message
     if spamCount > Section580.netSpamThreshold
         alertMessage = "Player likely spamming network messages! #{plyNick} (#{plySteamId}) is spamming: '#{strName}' (Count: #{spamCount} per #{Section580.netClearTime} seconds)"
         warnLog alertMessage
         Section580.Alerter\alertStaff plySteamId, plyNick, strName, "likely"
 
+        return true
 
 net.Incoming = ( len, client ) ->
-    len -= 16
-    strName = NetworkIDToString ReadHeader!
+    header = ReadHeader!
+    strName = NetworkIDToString header
 
     return unless strName
+    if strName == "nil"
+        warnLog "Invalid network message sent by '#{client}': Header: '#{header}' | strName: '#{strName} | len: #{len}' - ignoring"
+        -- return TODO: Decide if we should RERO here
```

Comment:
We likely still want to tally it, but it'll later get ignored by the `if not func` and trigger a double warn

Reply:
Weirdly enough, we don't see that functionality currently. When someone is actually spamming `nil` maliciously, only that alert triggers.

In either case, with `tallyUsage` instructing `net.Incoming` to stop processing a message, that shouldn't happen 👍